streams/src/test/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContextTest.java

```diff
@@ -155,29 +155,20 @@ public void shouldReturnEmptyHeadersIfHeadersAreNotSet() {
         assertThat(context.headers(), is(emptyIterable()));
     }
 
-    @Test
-    public void shouldThrowIllegalStateExceptionOnHeadersIfNoRecordContext() {
```

Comment:
This test was broken (missing `fail()` statement in the try-catch block).

The test should have been removed via https://github.com/apache/kafka/pull/9361 though.

Reply:
@ableegoldman Follow up though: should we keep this test using reverted logic: `shouldReturnEmptyHeaderIfRecordContextIsNull` ?

Reply:
Just want to make sure I understand the background here: as of KIP-478 we no longer throw if there's no active RecordContext, but instead just return empty. Is that right? If so then yes, an updated test like `shouldReturnEmptyHeaderIfRecordContextIsNull` seems good to have